packages/jazz-tools/src/tools/tests/coMap.test.ts

```diff
@@ -744,6 +744,81 @@ describe("CoMap", async () => {
     });
   });
 
+  describe("has", () => {
+    test("should return true if the key is defined", () => {
+      const Person = co.map({
+        name: z.string(),
+        age: z.number().optional(),
+      });
+
+      const person = Person.create({ name: "John", age: 20 });
+
+      expect(person.$jazz.has("name")).toBe(true);
+      expect(person.$jazz.has("age")).toBe(true);
+    });
+
+    test("should return false if the key is not defined", () => {
+      const Person = co.map({
+        name: z.string(),
+        age: z.number().optional(),
+      });
+
+      const person = Person.create({ name: "John" });
+
+      expect(person.$jazz.has("age")).toBe(false);
+    });
+
+    test("should not load the referenced CoValue", async () => {
+      const Person = co.map({
+        name: co.plainText(),
+      });
+
+      const { clientAccount, serverAccount } = await setupTwoNodes();
+
+      const person = Person.create(
+        {
+          name: "John",
+        },
+        { owner: Group.create(clientAccount).makePublic() },
+      );
+
+      await person.$jazz.waitForSync({ timeout: 1000 });
+
+      const loadedPerson = await Person.load(person.$jazz.id, {
+        resolve: true,
+        loadAs: serverAccount,
+      });
+
+      assert(loadedPerson);
+      expect(loadedPerson.$jazz.has("name")).toBe(true);
+      expect(loadedPerson.name).toBeNull();
```

Comment:
This check won't work, because the property access is able to load the value syncronously in the test environment.

The loadedPerson is also on a server peer, so the CoPlainText is sent directly when created, so it's already in the server node after you call `waitForSync`